### ISSUE TITLE
Template value throw type error when using nested property that is undefined

### ISSUE THREAD

**Issue by TTMoon** [sensedeep/dynamodb-onetable#476]
**Describe the bug**

Template value throw type error when using nested property that is undefined

**To Reproduce**

**Schema**
```
{
    version: '0.0.1',
    indexes: {
        primary: { hash: 'pk', sort: 'sk' },
        gsi1: { hash: 'gsi1pk', sort: 'gsi1sk' },
        gsi2: { hash: 'gsi2pk', sort: 'gsi2sk' },
        gsi3: { hash: 'gsi3pk', sort: 'gsi3sk' }
    },
    models: {
        Test: {
            pk: { type: String, value: 'test#${id}' },
            sk: { type: String, value: 'test#' },
            id: { type: String, generate: 'ulid' },

            user: {
                type: Object,
                schema: {
                    id: { type: String },
                    name: { type: String }
                }
            },

            gsi1pk: { type: String, value: 'user#${user.id}' },
            gsi1sk: { type: String, value: 'test#${id}' }
        }
    } as const,
    params: {
        timestamps: true
    }
};
```

**Code**
```
const client = new Dynamo({
    client: new DynamoDBClient()
});

const table = new Table({
    name: process.env.TABLE_NAME,
    client,
    schema,
    partial: true
});

const testClient = table.getModel<any>('Test');
const result = await testClient.create({});
```

**Error Stack**
```
TypeError: Cannot read properties of undefined (reading 'id')
at Model.getPropValue (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:263:18)
      at ../../node_modules/dynamodb-onetable/dist/cjs/Model.js:1601:26
          at String.replace (<anonymous>)
      at Model.runTemplate (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:1599:23)
      at Model.runTemplates (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:1583:38)
      at Model.collectProperties (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:1293:14)
      at Model.prepareProperties (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:1162:24)
      at Model.<anonymous> (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:955:35)
      at ../../node_modules/dynamodb-onetable/dist/cjs/Model.js:8:71
      at Object.<anonymous>.__awaiter (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:4:12)
      at Model.putItem (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:939:16)
      at Model.<anonymous> (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:532:37)
      at ../../node_modules/dynamodb-onetable/dist/cjs/Model.js:8:71
      at Object.<anonymous>.__awaiter (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:4:12)
      at Model.create (../../node_modules/dynamodb-onetable/dist/cjs/Model.js:523:16)
```
**Expected behavior**
The record should be created successfully.
```
{
        id: '01H2N36T21E6A9S21739EYXDBK',
        created: 2023-06-11T11:24:35.776Z,
        updated: 2023-06-11T11:24:35.776Z
}
```

In my opinion, we should handle undefined value on this line
[https://github.com/sensedeep/dynamodb-onetable/blob/main/src/Model.js#L273](https://github.com/sensedeep/dynamodb-onetable/blob/main/src/Model.js#L273)

**Environment (please complete the following information):**
 - OS Mac OS X
 - Node Version 18
 - OneTable Version 2.6.2
 - TypeScript Version 4.9.5

**Comment by mobsense** [comment]
Thank you for diagnosing thiss.

Can you please test for you if at that line you suggest, you return undefined?

**Comment by TTMoon** [comment]
I changed that line
```
getPropValue(properties, path) {
        let v = properties
        for (let part of path.split('.')) {
            if(v == null) return v
            v = v[part]
        }
        return v
}
```

 and run `npm test`
```
Test Suites: 52 passed, 52 total
Tests:       355 passed, 355 total
Snapshots:   0 total
Time:        95.154 s
```

and my case that nested property is undefined also works fine.

**Comment by mobsense** [comment]
Fix committed for 2.6.3
